addons/hr/tests/test_hr_employee.py

```diff
@@ -209,3 +209,25 @@ def test_employee_update_work_contact_id(self):
         employee_B.work_email = 'new_email@example.com'
         self.assertEqual(employee_A.work_email, 'employee_A@example.com')
         self.assertEqual(employee_B.work_email, 'new_email@example.com')
+
+    def test_change_user_on_employee(self):
+        partner = self.env['res.partner'].create({
+            'name' : 'Test User Partner'
+            })
+        bank_account = self.env['res.partner.bank'].create({
+            'acc_number' : '1234567',
+            'partner_id' : partner.id
+            })
```

Comment:
the test doesn't fail for me without the fix, I think the account should be on another company than `self.res_users_hr_officer` user to be able to reproduce it